### PR TITLE
Add tooltip displaying the exact time of the last collection item edit

### PR DESCRIPTION
**Pull request by kulyk**
Adds a tooltip displaying the exact time of the most recent edit, as the table only displays day, month, and year.

Also, this PR adds a `DateTime` component that formats date-time respecting the user's Metabase localization settings (stuff like 12h-clock vs. 24h-clock, date formats, etc.). I couldn't find an appropriate helper for that. ⚠️ There is a high chance I reinvented the wheel here. Please LMK if there is a helper for that already. If not, maybe it's worth checking the l10n settings are respected in different areas of the application.

A bit more on a11y:
* it's now possible to set an a11y role for a `Popover` component
* all tooltips will now have `role="tooltip"`

### To Verify

~⚠️ You may notice that the date and time do not update once you actually edit something. It's a known BE issue, and there is a PR for that already #16520~ (merged and rebased, should work as expected now)

1. Open a collection page
2. Hover any item's "last edited at" cell
3. A tooltip displaying a timestamp like "June 3, 2021, 10:12 AM" should appear

### Demo

Screenshot 2021-06-08 at 18 08 32

### CODE REVIEW

**Review by gusaiani — COMMENTED** [review]
As it is (16'' MBP screen), the tooltip is fairly disaligned from the original timestamp.

What if we made it look like this?
(could also be centered with the timestamp, and the up-arrow centered too…)

From:
 


To:

**Comment by kulyk** [comment]
> As it is (16'' MBP screen), the tooltip is fairly disaligned from the original timestamp.
> 
> What if we made it look like this?
> (could also be centered with the timestamp, and the up-arrow centered too…)
> 
> From:
> 
> 
> 
> To:
> 
> 

Wow, good catch! Will look into it, thanks

**Review by gusaiani — COMMENTED** [review]
I just fell into a bit of a rabbit hole here finding files with date and time formatters.

We have `frontend/src/metabase/lib/{time, formatting}.js`, both with formatting functions.

I wonder if you have checked what you can reuse from these and other files.

Also, there is [react-moment](https://www.npmjs.com/package/react-moment), which might be able to provide some of what you are coding for here. I've used it, it works nicely, but I don't remember details.

Finally there is the thinking about whether to use moment.js at all, since lighter libs are available...but that's a whole'nother story :)))

**Comment by kulyk** [comment]
> I just fell into a bit of a rabbit hole here finding files with date and time formatters.
> 
> We have `frontend/src/metabase/lib/{time, formatting}.js`, both with formatting functions.
> 
> I wonder if you have checked what you can reuse from these and other files.

Oh yeah, I've been there and felt kinda lost, to be honest. I found [`formatDateTime`](https://github.com/metabase/metabase/blob/e736e6b2956b14074ea37e0d70accb724812075d/frontend/src/metabase/lib/formatting.js#L485) function under the `/lib/formatting` and `DateTime` component uses `formatDateTimeWithUnit` (a more general version of it, `formatDateTime` just calls it, hardcoding one of the arguments).

Ideally, I was looking for something that takes user’s preferences from `MetabaseSettings` and calls something like `formatDateTime` with them. There is a function doing exactly that, but for some reason, it’s defined locally inside one of the query builder’s components (see [here](https://github.com/metabase/metabase/blob/e736e6b2956b14074ea37e0d70accb724812075d/frontend/src/metabase/query_builder/components/template_tags/CardTagEditor.jsx#L125))

That makes me think it could be a problem we haven't solved yet and the `DateTime` component is a possible solution. 


> Also, there is [react-moment](https://www.npmjs.com/package/react-moment), which might be able to provide some of what you are coding for here. I've used it, it works nicely, but I don't remember details.
> 
> Finally there is the thinking about whether to use moment.js at all, since lighter libs are available...but that's a whole'nother story :)))

I think we should definitely consider dropping `moment` at some point, so maybe it'd make sense to use our own abstraction, and perhaps it would be easier to swipe `moment` under the hood in the future

**Comment by gusaiani** [comment]
> There is a function doing exactly that, but for some reason, it’s defined locally inside one of the query builder’s components (see here)

This is one of the reasons I believe we should define these helpers not in the component files, but in `helper`-type files.

We seem to have a good many of them doing different things.

**Review by gusaiani — CHANGES_REQUESTED** [review]
Just noticed an unused file:

containers/dnd/PinPositionDropTarget.jsx

**Comment by gusaiani** [comment]
Sorry for the pulverized comments! 

e2e test?
Maybe even faster-running unit tests?

**Comment by kulyk** [comment]
> Sorry for the pulverized comments!
> 
> e2e test?
> Maybe even faster-running unit tests?

No problem, check out [table tests](https://github.com/metabase/metabase/pull/16492/files?file-filters%5B%5D=.js#diff-7efa1a785ffbe9aaa3b4599f125f493c0b06e5e7b4e4ceb5d2d9852170eb76d1) and [datetime tests](https://github.com/metabase/metabase/pull/16492/files?file-filters%5B%5D=.js#diff-b034d0c202439227c9b1a2809510930d0519cafd1bcfc75d73945cdea2ac7c9a)

**Comment by gusaiani** [comment]
> No problem, check out table tests and datetime tests

Oh yeah sure! Too much context-switching here. Thank you!

**Comment by kulyk** [comment]
@gusaiani can you please pull the branch once again and check if the tooltip alignment is better now?

File: frontend/src/metabase/components/DateTime.jsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    PropTypes.string,
+    PropTypes.instanceOf(Date),
+    PropTypes.number, // UNIX timestamp
+  ]).isRequired,
+  unit: PropTypes.oneOf(DATE_TIME_UNITS),
+};
+
+function DateTime({ value, unit = "default", ...props }) {
+  const options = MetabaseSettings.formattingOptions();
+  return <span {...props}>{formatDateTimeWithUnit(value, unit, options)}</span>;
```
Comment (gusaiani) [inline frontend/src/metabase/components/DateTime.jsx:40]: Slightly more readable?
💅

```suggestion
  const formattedTime = formatDateTimeWithUnit(value, unit, options);
  
  return <span {...props}>{formattedTime}</span>;
```

File: frontend/src/metabase/lib/settings.js [hunk 2]
```diff
@@ -121,6 +121,11 @@ class Settings {
     return this.get("anon-tracking-enabled") || false;
   }
 
+  formattingOptions() {
+    const opts = this.get("custom-formatting");
+    return opts && opts["type/Temporal"] ? opts["type/Temporal"] : {};
```
Comment (gusaiani) [inline frontend/src/metabase/lib/settings.js:126]: Sometimes I miss my optional chaining :)

**Review by gusaiani — APPROVED** [review]
Working nicely 👍

Might benefit from a look by someone experienced with our codebase (to maybe point to existing code we can reuse).

A minor comment inlined.

All in all, looks mergeable to me, good stuff Anton 👍

File: frontend/src/metabase/components/DateTime.jsx (hunk 1 above)
Comment (kulyk) [reply frontend/src/metabase/components/DateTime.jsx:40]: Yep!

File: frontend/src/metabase/lib/settings.js (hunk 2 above)
Comment (kulyk) [reply frontend/src/metabase/lib/settings.js:126]: Oh yeah, I feel it 😄

**Comment by kulyk** [comment]
Merging it now. Feel free to leave a post-merge review or revisit that when the feature branch is going to be merged

**Pull request merged**